src/sacn/merge_receiver.c

```diff
@@ -604,12 +603,114 @@ etcpal_error_t sacn_merge_receiver_get_source_cid(sacn_merge_receiver_t handle,
 void universe_data(sacn_receiver_t handle, const EtcPalSockAddr* source_addr, const SacnHeaderData* header,
                    const uint8_t* pdata, bool is_sampling, void* context)
 {
-  ETCPAL_UNUSED_ARG(handle);
-  ETCPAL_UNUSED_ARG(source_addr);
-  ETCPAL_UNUSED_ARG(header);
-  ETCPAL_UNUSED_ARG(pdata);
-  ETCPAL_UNUSED_ARG(is_sampling);
   ETCPAL_UNUSED_ARG(context);
+
+  // Look up merger/source info & update pending status if source exists.
+  sacn_source_id_t source_id = SACN_DMX_MERGER_SOURCE_INVALID;
+  sacn_dmx_merger_t merger_handle = SACN_DMX_MERGER_INVALID;
+  bool use_pap = false;
+  if (sacn_lock())
+  {
+    SacnMergeReceiver* merge_receiver = NULL;
+    if (lookup_merge_receiver((sacn_merge_receiver_t)handle, &merge_receiver, NULL) == kEtcPalErrOk)
+    {
+      merger_handle = merge_receiver->merger_handle;
+      use_pap = merge_receiver->use_pap;
+
+      SacnMergeReceiverSource* source = NULL;
+      if (lookup_merge_receiver_source(merge_receiver, &header->cid, &source) == kEtcPalErrOk)
+      {
+        source_id = source->id;
+
+        if (source->pending && (header->start_code == 0x00))
+        {
+          source->pending = false;
+          --merge_receiver->num_pending_sources;
+        }
+      }
+    }
+
+    sacn_unlock();
+  }
+
+  // If this source is new, add it.
+  bool new_source = false;
+  if (source_id == SACN_DMX_MERGER_SOURCE_INVALID)
+  {
+    sacn_dmx_merger_add_source(merger_handle, &source_id);
+    new_source = true;
+  }
+
+  if (new_source && sacn_lock())
+  {
+    SacnMergeReceiver* merge_receiver = NULL;
+    if (lookup_merge_receiver((sacn_merge_receiver_t)handle, &merge_receiver, NULL) == kEtcPalErrOk)
+      add_sacn_merge_receiver_source(merge_receiver, source_id, &header->cid, (header->start_code != 0x00));
+
+    sacn_unlock();
+  }
+
+  // Execute a new merge if needed.
+  bool new_merge = false;
```

Comment:
I think new_merge isn't quite the right name, because it's used to indicate that a new merge has taken place and a notification is needed.  I initially thought that the reader of the flag would execute the merge.  How about notify_merge?

Reply:
> it's used to indicate that a new merge has taken place and a notification is needed
new_merge = true doesn't always mean a notification is needed. The only purpose is to pass on the news that a merge occurred, which factors into the decision whether to notify or not. But to avoid the confusion of if the merge was already done, I'll rename this to new_merge_occurred.

Reply:
https://github.com/ETCLabs/sACN/commit/9ba306e80ec677c04fd7542bd466245671a5a55c